engine/access/collection_sync/fetcher/engine.go

```diff
@@ -0,0 +1,131 @@
+package fetcher
+
+import (
+	"fmt"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/engine"
+	"github.com/onflow/flow-go/engine/access/collection_sync"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/component"
+	"github.com/onflow/flow-go/module/irrecoverable"
+	"github.com/onflow/flow-go/module/jobqueue"
+	"github.com/onflow/flow-go/state/protocol"
+	"github.com/onflow/flow-go/storage"
+)
+
+// Fetcher is a component that consumes finalized block jobs and processes them
+// to index collections. It uses a job consumer with windowed throttling to prevent node overload.
+type Fetcher struct {
+	component.Component
+
+	consumer       *jobqueue.ComponentConsumer
+	blockProcessor collection_sync.BlockProcessor
+	workSignal     engine.Notifier
+	metrics        module.CollectionSyncMetrics
+}
+
+var _ collection_sync.Fetcher = (*Fetcher)(nil)
+var _ collection_sync.ProgressReader = (*Fetcher)(nil)
+var _ component.Component = (*Fetcher)(nil)
+
+// NewFetcher creates a new Fetcher component.
+//
+// Parameters:
+//   - log: Logger for the component
+//   - blockProcessor: BlockProcessor implementation for processing collection indexing jobs
+//   - progressInitializer: Initializer for tracking processed block heights
+//   - state: Protocol state for reading finalized block information
+//   - blocks: Blocks storage for reading blocks by height
+//   - maxProcessing: Maximum number of jobs to process concurrently
+//   - maxSearchAhead: Maximum number of jobs beyond processedIndex to process. 0 means no limit
+//   - metrics: Optional metrics collector for reporting collection fetched height
+//
+// No error returns are expected during normal operation.
+func NewFetcher(
+	log zerolog.Logger,
+	blockProcessor collection_sync.BlockProcessor, progressInitializer storage.ConsumerProgressInitializer,
+	state protocol.State,
+	blocks storage.Blocks,
+	maxProcessing uint64, // max number of blocks to fetch collections
+	maxSearchAhead uint64, // max number of blocks beyond the next unfullfilled height to fetch collections for
+	metrics module.CollectionSyncMetrics, // optional metrics collector
+) (*Fetcher, error) {
+	workSignal := engine.NewNotifier()
+
+	// Read the default index from the finalized root height
+	defaultIndex := state.Params().FinalizedRoot().Height
```

Comment:
Shouldn't this be the sealed root block? Otherwise, if the node was dynamic bootstrapped, we'd never get collections for the initial unsealed finalized blocks
```suggestion
	defaultIndex := state.Params().SealedRoot().Height
```